### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by EmiliaCiastek**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,11 +8,12 @@
     <artifactId>statki</artifactId>
     <packaging>pom</packaging>
     <version>1.0-SNAPSHOT</version>
-    <url>https://github.com/EmiliaCiastek/Ships</url>
+    <url>https://github.com/patiem/ships_JA</url>
```
Comment (szczepanskikrs) [inline pom.xml]: Will this work when you generate documentation with MVN site?

File: pom.xml [hunk 2]
```diff
@@ -83,21 +82,17 @@
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
-
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-core</artifactId>
-            <version>2.9.2</version>
+            <version>2.9.3</version>
```
Comment (szczepanskikrs) [inline pom.xml]: Hardcoded dependencies! Use properties!!!

File: pom.xml [hunk 3]
```diff
@@ -83,21 +82,17 @@
...
         </dependency>
-
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
-            <version>2.6.0</version>
+            <version>2.9.3</version>
```
Comment (szczepanskikrs) [inline pom.xml]: Hardcoded dependencies! Use properties!!!

File: server/src/main/java/communication/Position.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package communication;
+
+class Position {
+
+    private int index;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Position.java]: final

File: server/src/main/java/communication/Position.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    private int column;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Position.java]: final

File: server/src/main/java/communication/Position.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    private int row;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Position.java]: final

File: server/src/main/java/communication/Player.java [hunk 7]
```diff
@@ -3,9 +3,6 @@
 import fleet.Fleet;
 import fleet.HardcodedFleet;
 
-import java.io.IOException;
-import java.io.OutputStreamWriter;
-import java.io.PrintWriter;
 import java.net.Socket;
 
 public class Player {
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Player.java]: Must be public?

File: server/src/main/java/communication/PositionAdapter.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package communication;
+
+class PositionAdapter {
+
+    static Position createPositionFromCoordinates(int column, int row) {
+        int index = row * 10 + column;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/PositionAdapter.java]: Magic number!

File: server/src/main/java/communication/PositionAdapter.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+        return new Position(index);
+    }
+
+    static Position createPositionFromIndex(int providedIndex) {
+        int column = providedIndex % 10;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/PositionAdapter.java]: Magic number

File: server/src/main/java/communication/PositionAdapter.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        int row = providedIndex / 10;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/PositionAdapter.java]: Magic number

File: server/src/main/java/fleet/CustomFleet.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
+package fleet;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CustomFleet implements Fleet {
+    private List<Ship> ships;
```
Comment (szczepanskikrs) [inline server/src/main/java/fleet/CustomFleet.java]: Can it be final?

File: server/src/main/java/fleet/CustomFleet.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    protected List<Integer> fleetPositions;
```
Comment (szczepanskikrs) [inline server/src/main/java/fleet/CustomFleet.java]: Can it be final?

File: server/src/main/java/fleet/CustomFleet.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    private List<Integer> hitFields;
```
Comment (szczepanskikrs) [inline server/src/main/java/fleet/CustomFleet.java]: Can it be final?

File: server/src/main/java/fleet/CustomFleet.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+    public List<Integer> getFleetPositions() {
+        ships.stream()
+                .forEach(ship -> fleetPositions.addAll(ship.getFields()));
+
+        return fleetPositions;
+    }
+
+    public void hit(int position) {
+    } //TODO: to be implemented
```
Comment (szczepanskikrs) [inline server/src/main/java/fleet/CustomFleet.java]: Should throw NotImplementedException

File: server/src/main/java/fleet/HardcodedFleet.java [hunk 15]
```diff
@@ -30,4 +30,14 @@ public void hit(int position) {
     public List<Integer> getHitFields() {
         return hitFields;
     }
+
+    @Override
+    public int getSize() {
+        return 0;
```
Comment (szczepanskikrs) [inline server/src/main/java/fleet/HardcodedFleet.java]: Hardcoded magic number

File: server/src/main/java/fleet/PositionValidator.java [hunk 16]
```diff
@@ -3,16 +3,16 @@
 class PositionValidator {
     private static final int MIN_POSITION_INDEX = 0;
     private static final int MAX_POSITION_INDEX = 99;
-    private Fleet fleet;
+    private CustomFleet fleet;
```
Comment (szczepanskikrs) [inline server/src/main/java/fleet/PositionValidator.java]: Final?

File: utils/pom.xml [hunk 17]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>statki</artifactId>
+        <groupId>com.javacademy</groupId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
```
Comment (szczepanskikrs) [inline utils/pom.xml]: Hardcoded use properties...

File: utils/src/main/java/JsonGenerator.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+
+public class JsonGenerator {
+    String createJson(Object object) {
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonString = "";
+
+        try {
+            jsonString = mapper.writeValueAsString(object);
+            //TODO: this.objectMapper.setVisibility(PropertyAccessor.SETTER, JsonAutoDetect.Visibility.NON_PRIVATE);
```
Comment (szczepanskikrs) [inline utils/src/main/java/JsonGenerator.java]: Fix this

File: utils/src/main/java/JsonGenerator.java [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+
+        } catch (JsonProcessingException e) {
+            e.printStackTrace(); //TODO : exception handler
```
Comment (szczepanskikrs) [inline utils/src/main/java/JsonGenerator.java]: Logger?

File: utils/src/test/java/DummyObject.java [hunk 20]
```diff
@@ -0,0 +1,34 @@
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+public class DummyObject {
+    //class used only for Json testing
```
Comment (szczepanskikrs) [inline utils/src/test/java/DummyObject.java]: Dangling comment

File: pom.xml [hunk 21]
```diff
@@ -83,21 +82,17 @@
             <version>1.10.19</version>
```
Comment (szczepanskikrs) [inline pom.xml]: Hardcoded value! Use properties

File: utils/pom.xml (hunk 17 above)
Comment (EmiliaCiastek) [reply utils/pom.xml]: modelVersion can't be a property

File: utils/src/test/java/DummyObject.java (hunk 20 above)
Comment (EmiliaCiastek) [reply utils/src/test/java/DummyObject.java]: I've replaced this ugly comment with JavaDoc comment.
Not every mock has to be created for example by Mockito. It'a allowed sometimes to use fake classes (look at Chapter 3 in "Working Effectively with Legacy Code" by Michael Feathers).

File: server/src/main/java/communication/Player.java [hunk 22]
```diff
@@ -5,27 +5,27 @@
 
 import java.net.Socket;
 
-public class Player {
+class Player {
 
     private String name;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Player.java]: Final

File: server/src/main/java/communication/Player.java [hunk 23]
```diff
@@ -5,27 +5,27 @@
...
     private Socket socket;
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Player.java]: Final?

**Pull request merged**